cmd/vdev_id/vdev_id

```diff
@@ -140,7 +140,7 @@ Usage: vdev_id [-h]
   -p    number of phy's per switch port [default=$PHYS_PER_PORT]
   -h    show this summary
 EOF
-	exit 0
+	exit 1
```

Comment:
`-h` is a valid flag and should return an error code

Reply:
An error code on -h (assuming -h is specified by a human on the shell command) will not hurt anyone. I can't image -h is used in a udev rule or script on purpose.
However, if -h is mistyped (-j for instance is the next key on the keyboard) in a udev rule, /dev will be full of funny links or other similar consequences of processing the usage message where real data are expected. 

If it weren't a udev helper I'd never proposed to return error on a usage call.

Reply:
Sorry, I meant to say "-h is a valid flag and should **not** return an error code".  A non-valid argument like `-j` should return an error code.  I think the problem is that `usage()` is returning an exit code, when all it should be doing is print our the usage message.

Reply:
-j is a valid flag for vdev_id, according to usage message. 
The process cannot terminate w/o returning an exit code. According to https://tldp.org/LDP/abs/html/exit-status.html exiting w/o a code will return the exit code of the last executed program - i.e. we  relinquishing control of the exit code and may have a situation when we exit with zero code, however the stdout is not suitable for consumption by the udev rule. 

My suggestion is to use the exit code as the clear indicator of the usability of the stdout for a udev rule.
See https://www.freedesktop.org/software/systemd/man/udev.html#PROGRAM for the logic of the udev rule processing. The match is triggered only when we return SUCCESS (i.e. zero). For an udev rule, -h invocation can never be a success.

Reply:
Ok, maybe I'm being too pedantic with `-h` returning 1.  You can leave it as-is